source/MaterialXGenShader/ColorManagementSystem.cpp

```diff
@@ -0,0 +1,94 @@
+#include <MaterialXGenShader/ColorManagementSystem.h>
+
+#include <MaterialXGenShader/ShaderGenerator.h>
+#include <MaterialXGenShader/Nodes/SourceCodeNode.h>
+
+namespace MaterialX
+{
+
+ColorSpaceTransform::ColorSpaceTransform(const string& ss, const string& ts, const TypeDesc* t)
+    : sourceSpace(ss)
+    , targetSpace(ts)
+    , type(t)
+{
+    if (type != Type::COLOR3 && type != Type::COLOR4)
+    {
+        throw ExceptionShaderGenError("Color space transform can only be a color3 or color4.");
+    }
+}
+
+
+ColorManagementSystem::ColorManagementSystem(ShaderGenerator& shadergen, const string& configFile)
+    : _configFile(configFile)
+    , _shadergen(shadergen)
+{
+}
+
+bool ColorManagementSystem::registerImplementation(const ColorSpaceTransform& transform, CreatorFunction<ShaderNodeImpl> creator)
+{
+    if (transform.type)
+    {
+        string implName = getImplementationName(transform);
+        _implFactory.registerClass(implName, creator);
+        _registeredImplNames.push_back(implName);
+        return true;
+    }
+    return false;
+}
+
+void  ColorManagementSystem::setConfigFile(const string& configFile)
+{
+    _configFile = configFile;
+    _implFactory.unregisterClasses(_registeredImplNames);
+    _registeredImplNames.clear();
+    _cachedImpls.clear();
+}
+
+void ColorManagementSystem::loadLibrary(DocumentPtr document)
+{
+    _document = document;
+    _implFactory.unregisterClasses(_registeredImplNames);
+    _registeredImplNames.clear();
+    _cachedImpls.clear();
```

Comment:
The docs for this method should say that any previously loaded content will be replaced. Or you could change it to support multiple loads. In that case keeping an internally created document that gets appended with the given document contents. Maybe overkill for now though, and a docs update is easier :)